packages/components/tabs/src/TabsList.tsx

```diff
@@ -0,0 +1,112 @@
+import * as RadixTabs from '@radix-ui/react-tabs'
+import { Button } from '@spark-ui/button'
+import { Icon } from '@spark-ui/icon'
+import { ArrowVerticalLeft, ArrowVerticalRight } from '@spark-ui/icons'
+import { forwardRef, type ReactElement, useEffect, useMemo, useRef, useState } from 'react'
+
+import { useTabsContext } from './TabsContext'
+import { listStyles, navigationArrowStyles, wrapperStyles } from './TabsList.styles'
+
+export type TabsListProps = Omit<RadixTabs.TabsListProps, 'children'> & {
+  children: ReactElement[]
+}
+
+const getSelectedTabIndex = (list: ReactElement[], selectedTab?: string) => {
+  const matchingChild = list.find(elm => (elm?.props.value as string) === selectedTab)
+
+  return matchingChild ? list.indexOf(matchingChild) : 0
+}
+
+export const TabsList = forwardRef<HTMLDivElement, TabsListProps>(
+  (
+    {
+      /**
+       * Default Radix Primitive values
+       * see https://www.radix-ui.com/docs/primitives/components/tabs#list
+       */
+      asChild = false,
+      loop = true,
+      children,
+      ...rest
+    },
+    ref
+  ) => {
+    const innerRef = useRef(null)
+    const listRef = ref || innerRef
+    const { orientation, selectedTab, setSelectedTab } = useTabsContext()
+    const tabs = useRef<ReactElement[]>(children)
+
+    const tabPivots = useMemo(() => {
```

Comment:
Since the computation done inside this useMemo does not seem expensive, and `selectedTab` is a dependency, is it really necessary to memoize this function?